Run testcase  failed on windows

I found some test case will fail on windows.
https://github.com/helm/helm/blob/2e3e22a0033cc6d3833d6d3e84353c1aecb06dc6/pkg/chartutil/save_test.go#L37-L45
https://github.com/helm/helm/blob/2e3e22a0033cc6d3833d6d3e84353c1aecb06dc6/pkg/chartutil/save_test.go#L200-L204
output like below:
```
=== RUN   TestSaveDir
    save_test.go:230: Templates data did not match
--- FAIL: TestSaveDir (25.29s)
```

One of the reason why first testcase failed is that `path.Join(tmp, "newdir")` should be `filepath.Join` in line 41,the second is that variable `c2` should be compared to `c` and its Templates's name is always combined with slashes, not applicable to `filepath.Join`

--- a/pkg/chartutil/save_test.go
+++ b/pkg/chartutil/save_test.go
@@ -38,7 +38,7 @@ func TestSave(t *testing.T) {
 	tmp := ensure.TempDir(t)
 	defer os.RemoveAll(tmp)
 
-	for _, dest := range []string{tmp, path.Join(tmp, "newdir")} {
+	for _, dest := range []string{tmp, filepath.Join(tmp, "newdir")} {
 		t.Run("outDir="+dest, func(t *testing.T) {
 			c := &chart.Chart{
 				Metadata: &chart.Metadata{
@@ -210,7 +210,7 @@ func TestSaveDir(t *testing.T) {
 			{Name: "scheherazade/shahryar.txt", Data: []byte("1,001 Nights")},
 		},
 		Templates: []*chart.File{
-			{Name: filepath.Join(TemplatesDir, "nested", "dir", "thing.yaml"), Data: []byte("abc: {{ .Values.abc }}")},
+			{Name: path.Join(TemplatesDir, "nested", "dir", "thing.yaml"), Data: []byte("abc: {{ .Values.abc }}")},
 		},
 	}
 
@@ -227,11 +227,11 @@ func TestSaveDir(t *testing.T) {
 		t.Fatalf("Expected chart archive to have %q, got %q", c.Name(), c2.Name())
 	}
 
-	if len(c2.Templates) != 1 || c2.Templates[0].Name != filepath.Join(TemplatesDir, "nested", "dir", "thing.yaml") {
+	if len(c2.Templates) != 1 || c2.Templates[0].Name != c.Templates[0].Name {
 		t.Fatal("Templates data did not match")
 	}
 
-	if len(c2.Files) != 1 || c2.Files[0].Name != "scheherazade/shahryar.txt" {
+	if len(c2.Files) != 1 || c2.Files[0].Name != c.Files[0].Name {
 		t.Fatal("Files data did not match")
 	}
 }

